solr/solrj/src/java/org/apache/solr/client/solrj/impl/LBHttpSolrClient.java

```diff
@@ -122,10 +122,14 @@ protected HttpSolrClient makeSolrClient(String server) {
     HttpSolrClient client;
     if (httpSolrClientBuilder != null) {
       synchronized (this) {
-        httpSolrClientBuilder.withBaseSolrUrl(server).withHttpClient(httpClient);
-        httpSolrClientBuilder.withConnectionTimeout(connectionTimeoutMillis, TimeUnit.MILLISECONDS);
-        httpSolrClientBuilder.withSocketTimeout(soTimeoutMillis, TimeUnit.MILLISECONDS);
-
+        httpSolrClientBuilder
+            .withBaseSolrUrl(server)
+            .withHttpClient(httpClient)
+            .withConnectionTimeout(connectionTimeoutMillis, TimeUnit.MILLISECONDS)
+            .withSocketTimeout(soTimeoutMillis, TimeUnit.MILLISECONDS);
+        if (defaultCollection != null) {
```

Comment:
would this have worked if you just blinding passed in `defaultCollection` and it happened to be null?   I/e with out this check?  Though maybe it's clearer to see the logic with the excplicit check...

Reply:
> would this have worked if you just blindly passed in `defaultCollection` and it happened to be null?

Yes and no.

The `withDefaultCollection` setter is resilient enough to handle null-values.  So in the general case that'd be just fine.

But the reason we check explicitly in this case is because a user might have specified a default-coll on their httpSolrClientBuilder (but not on the LB Builder directly).  In that scenario we'd be overwriting the value they provided with 'null'.

You'll notice this is the same pattern we follow with request-writer and a few other objects here - we're essentially just doing out best to make sure that we don't accidentally clobber values that were provided on the HttpSolrClientBuilder.